utils/datasets/cifar10_dataset.py

```diff
@@ -0,0 +1,330 @@
+# Copyright 2019, Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Library for loading and preprocessing CIFAR-10 training and testing data."""
+
+import collections
+from typing import Tuple
+import numpy as np
+import tensorflow as tf
+import tensorflow_federated as tff
+from tensorflow.keras import datasets, layers, models
+
+CIFAR_SHAPE = (32, 32, 3)
+TOTAL_FEATURE_SIZE = 32 * 32 * 3
+NUM_EXAMPLES_PER_CLIENT = 5000
+
+def load_cifar10_federated(
```

Comment:
It would be great to give an optional random seed governing any randomness contained, in order to make things fully reproducible.

I think this can be done via numpy.random.set_state. I'm okay with it in a follow-up, but I think it should be done before experiments start in earnest.